<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1068</width>
    <height>793</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QFrame" name="statusBarFrame">
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>211</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Time (s)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="statTime">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>FPS</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="statFps">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>50</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>VSYNC (F3)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="statVsync">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>16777215</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="mainTab">
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1068</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuRecentWorkspaces">
     <property name="title">
      <string>Recent Workspaces</string>
     </property>
    </widget>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionNewWorkspace"/>
    <addaction name="actionLoadWorkspace"/>
    <addaction name="menuRecentWorkspaces"/>
    <addaction name="separator"/>
    <addaction name="actionCloseWorkspace"/>
    <addaction name="separator"/>
    <addaction name="actionSaveWorkspace"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Window</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuEvent_log">
     <property name="title">
      <string>Event log</string>
     </property>
     <addaction name="actionLogShowInfo"/>
     <addaction name="actionLogShowWarning"/>
     <addaction name="actionLogShowError"/>
    </widget>
    <addaction name="actionViewToolbar"/>
    <addaction name="actionViewStatusbar"/>
    <addaction name="actionViewEventlog"/>
    <addaction name="actionViewWorkspace"/>
    <addaction name="separator"/>
    <addaction name="menuEvent_log"/>
   </widget>
   <widget class="QMenu" name="menuTemp">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Temp</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionUndo"/>
    <addaction name="separator"/>
    <addaction name="actionZoomIn"/>
    <addaction name="actionZoomOut"/>
    <addaction name="separator"/>
    <addaction name="actionReloadShaders"/>
    <addaction name="actionReloadTextures"/>
   </widget>
   <widget class="QMenu" name="menuWorkspace">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Workspace</string>
    </property>
    <addaction name="actionNewMaterial"/>
    <addaction name="actionNewParticleSystem"/>
    <addaction name="actionNewCustomShape"/>
    <addaction name="actionNewEntity"/>
    <addaction name="actionNewScene"/>
    <addaction name="actionNewScript"/>
    <addaction name="actionNewResource"/>
    <addaction name="separator"/>
    <addaction name="actionSelectResourceForEditing"/>
    <addaction name="separator"/>
    <addaction name="actionPackageResources"/>
    <addaction name="actionProjectSettings"/>
    <addaction name="separator"/>
    <addaction name="actionProjectPlay"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="actionImagePacker"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuWorkspace"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTemp"/>
   <addaction name="menuTools"/>
   <addaction name="menuView"/>
   <addaction name="menuWindow"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>26</width>
     <height>26</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QDockWidget" name="eventlogDock">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable|QDockWidget::DockWidgetVerticalTitleBar</set>
   </property>
   <property name="windowTitle">
    <string>Event log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QListView" name="eventlist">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAsNeeded</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="workspaceDock">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable|QDockWidget::DockWidgetVerticalTitleBar</set>
   </property>
   <property name="windowTitle">
    <string>Workspace resources</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTableView" name="workspace">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="showGrid">
        <bool>false</bool>
       </property>
       <property name="gridStyle">
        <enum>Qt::SolidLine</enum>
       </property>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderCascadingSectionResizes">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>21</number>
       </attribute>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/app_exit.png</normaloff>:/16x16_ico_png/app_exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionWindowClose">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/app_close_window.png</normaloff>:/16x16_ico_png/app_close_window.png</iconset>
   </property>
   <property name="text">
    <string>Close </string>
   </property>
   <property name="toolTip">
    <string>Close current window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionWindowNext">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/app_next_window.png</normaloff>:/16x16_ico_png/app_next_window.png</iconset>
   </property>
   <property name="text">
    <string>Next</string>
   </property>
   <property name="toolTip">
    <string>Swap to next window</string>
   </property>
  </action>
  <action name="actionWindowPrev">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/app_prev_window.png</normaloff>:/16x16_ico_png/app_prev_window.png</iconset>
   </property>
   <property name="text">
    <string>Prev</string>
   </property>
   <property name="toolTip">
    <string>Swap to previous window</string>
   </property>
  </action>
  <action name="actionViewToolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toolbar</string>
   </property>
  </action>
  <action name="actionViewStatusbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Statusbar</string>
   </property>
  </action>
  <action name="actionZoomIn">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/zoom_in.png</normaloff>:/16x16_ico_png/zoom_in.png</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl++</string>
   </property>
  </action>
  <action name="actionZoomOut">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/zoom_out.png</normaloff>:/16x16_ico_png/zoom_out.png</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionReloadShaders">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/refresh.png</normaloff>:/16x16_ico_png/refresh.png</iconset>
   </property>
   <property name="text">
    <string>Reload Shaders</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+S</string>
   </property>
  </action>
  <action name="actionViewEventlog">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Eventlog</string>
   </property>
  </action>
  <action name="actionViewWorkspace">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Workspace</string>
   </property>
  </action>
  <action name="actionDeleteResource">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/delete.png</normaloff>:/16x16_ico_png/delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="toolTip">
    <string>Delete resource object</string>
   </property>
  </action>
  <action name="actionNewMaterial">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New Material</string>
   </property>
   <property name="toolTip">
    <string>Create new material</string>
   </property>
  </action>
  <action name="actionNewParticleSystem">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New Particle System</string>
   </property>
  </action>
  <action name="actionEditResource">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/edit.png</normaloff>:/16x16_ico_png/edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit Resource</string>
   </property>
   <property name="toolTip">
    <string>Edit Resource</string>
   </property>
  </action>
  <action name="actionSaveWorkspace">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/workspace_save.png</normaloff>:/16x16_ico_png/workspace_save.png</iconset>
   </property>
   <property name="text">
    <string>Save Workspace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionEditResourceNewWindow">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/edit.png</normaloff>:/16x16_ico_png/edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit Resource in Win</string>
   </property>
   <property name="toolTip">
    <string>Edit Resource</string>
   </property>
  </action>
  <action name="actionEditResourceNewTab">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/edit.png</normaloff>:/16x16_ico_png/edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit Resource in Tab</string>
   </property>
   <property name="toolTip">
    <string>Edit Resource</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/settings.png</normaloff>:/16x16_ico_png/settings.png</iconset>
   </property>
   <property name="text">
    <string>Settings ...</string>
   </property>
  </action>
  <action name="actionReloadTextures">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/refresh.png</normaloff>:/16x16_ico_png/refresh.png</iconset>
   </property>
   <property name="text">
    <string>Reload Textures</string>
   </property>
   <property name="shortcut">
    <string>Alt+Shift+T</string>
   </property>
  </action>
  <action name="actionImagePacker">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/package.png</normaloff>:/16x16_ico_png/package.png</iconset>
   </property>
   <property name="text">
    <string>Image Packer ...</string>
   </property>
  </action>
  <action name="actionLoadWorkspace">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/workspace_open.png</normaloff>:/16x16_ico_png/workspace_open.png</iconset>
   </property>
   <property name="text">
    <string>Open Workspace ...</string>
   </property>
  </action>
  <action name="actionNewWorkspace">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/workspace_new.png</normaloff>:/16x16_ico_png/workspace_new.png</iconset>
   </property>
   <property name="text">
    <string>New Workspace ...</string>
   </property>
  </action>
  <action name="actionCloseWorkspace">
   <property name="text">
    <string>Close Workspace</string>
   </property>
  </action>
  <action name="actionPackageResources">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/package.png</normaloff>:/16x16_ico_png/package.png</iconset>
   </property>
   <property name="text">
    <string>Package ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+P</string>
   </property>
  </action>
  <action name="actionSelectResourceForEditing">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/edit.png</normaloff>:/16x16_ico_png/edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit Resource ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionNewResource">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionProjectSettings">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/settings.png</normaloff>:/16x16_ico_png/settings.png</iconset>
   </property>
   <property name="text">
    <string>Settings ...</string>
   </property>
   <property name="toolTip">
    <string>Workspace settings</string>
   </property>
  </action>
  <action name="actionProjectPlay">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/32x32_ico_png/play.png</normaloff>:/32x32_ico_png/play.png</iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionNewCustomShape">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New Shape</string>
   </property>
  </action>
  <action name="actionWindowPopOut">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/app_pop_out_window.png</normaloff>:/16x16_ico_png/app_pop_out_window.png</iconset>
   </property>
   <property name="text">
    <string>Pop Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+O</string>
   </property>
  </action>
  <action name="actionDuplicateResource">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/copy.png</normaloff>:/16x16_ico_png/copy.png</iconset>
   </property>
   <property name="text">
    <string>Duplicate</string>
   </property>
  </action>
  <action name="actionNewEntity">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New Entity</string>
   </property>
   <property name="toolTip">
    <string> New entity</string>
   </property>
  </action>
  <action name="actionClearLog">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/clear.png</normaloff>:/16x16_ico_png/clear.png</iconset>
   </property>
   <property name="text">
    <string>Clear Log</string>
   </property>
   <property name="toolTip">
    <string>Clear log events</string>
   </property>
  </action>
  <action name="actionLogShowInfo">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/log_info.png</normaloff>:/16x16_ico_png/log_info.png</iconset>
   </property>
   <property name="text">
    <string>Show Info</string>
   </property>
  </action>
  <action name="actionLogShowWarning">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/log_warning.png</normaloff>:/16x16_ico_png/log_warning.png</iconset>
   </property>
   <property name="text">
    <string>Show Warnings</string>
   </property>
  </action>
  <action name="actionLogShowError">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/log_error.png</normaloff>:/16x16_ico_png/log_error.png</iconset>
   </property>
   <property name="text">
    <string>Show Errors</string>
   </property>
  </action>
  <action name="actionNewScene">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New Scene</string>
   </property>
   <property name="toolTip">
    <string>New scene</string>
   </property>
  </action>
  <action name="actionNewScript">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/add.png</normaloff>:/16x16_ico_png/add.png</iconset>
   </property>
   <property name="text">
    <string>New Script</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/cut.png</normaloff>:/16x16_ico_png/cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="toolTip">
    <string>Cut the current object and place into clipboard</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/copy.png</normaloff>:/16x16_ico_png/copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="toolTip">
    <string>Copy the currently selected object to clipboard</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/paste.png</normaloff>:/16x16_ico_png/paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="toolTip">
    <string>Paste the current clipboard object</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="../resource/resource.qrc">
     <normaloff>:/16x16_ico_png/undo.png</normaloff>:/16x16_ico_png/undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="toolTip">
    <string>Undo the last action</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../resource/resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionViewToolbar</sender>
   <signal>toggled(bool)</signal>
   <receiver>mainToolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionViewStatusbar</sender>
   <signal>toggled(bool)</signal>
   <receiver>statusbar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>706</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionViewEventlog</sender>
   <signal>toggled(bool)</signal>
   <receiver>eventlogDock</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>533</x>
     <y>715</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionViewWorkspace</sender>
   <signal>toggled(bool)</signal>
   <receiver>workspaceDock</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
